### DISCUSSION TITLE
Only sort in one direction (or change the first-click sort direction)

### DISCUSSION THREAD

**Discussion by logcabintech** [rappasoft/laravel-livewire-tables#1003] [Q&A] [answered]
Is there a way to either:

1. make the first click on a header sort descending
2. remove ascending from the sort directions

Reason: columns that have no importance sorting in ascending order (last seen, last purchase). We always sort these columns in descending order, and never sort them in ascending order. This would save the second click on the header.

**Comment by lrljoe** [comment, marked as answer]
This will prevent a column "last_seen" from being sorted in ascending order, by always causing it to sort in a descending manner, or else be unsorted.

```
Column::make("Last Seen", 'last_seen')
        ->sortable(
            fn(Builder $query, string $direction) => $query->orderBy('last_seen', "desc")
),
```

**Reply by dirtbikr** [reply]
That results in a type error for me:

App\Http\Livewire\UserTable::App\Http\Livewire\{closure}(): Argument #1 ($query) must be of type App\Http\Livewire\Builder, Illuminate\Database\Eloquent\Builder given

EDIT
Adding `use Illuminate\Database\Eloquent\Builder` to the includes fixes the type error.

**Reply by logcabintech** [reply]
Thanks a lot. That does lock the order to descending. I also set `->setSortingPillDirections('Descending', 'Descending')` so the pill shows "Descending" regardless of the sort direction. Not the prettiest, but functional. I appreciate it!

**Reply by lrljoe** [reply]
I'll look at a core implementation for singleDirectionSort but no promises on timescales
